app/Database/PostInsertion.hs

```diff
@@ -7,18 +7,17 @@ module Database.PostInsertion(
 import qualified Data.Text as T
 import Config (databasePath)
 import Database.Persist.Sqlite (insert_, runSqlite, runMigration, SqlPersistM)
-import Database.Tables
+import Database.Tables hiding (postCode, description, postCategoryName)
 
 -- | Insert a new post into the database
 insertPost :: T.Text -> T.Text -> T.Text -> T.Text -> IO ()
-insertPost departmentName postName postCode description =
+insertPost departmentName1 postName0 postCode description =
```

Comment:
In general, don't append a number; instead, append an underscore (e.g., `departmentName_`).